### PR TITLE
Added foldr for tensorflow frontend

### LINKED ISSUE #21965: foldr

### PR DESCRIPTION
**Pull request by AbdullahSabry**
Closes #21965

### CODE REVIEW

**Comment by AbdullahSabry** [comment]
All the tests run successfully!

**Comment by AbdullahSabry** [comment]
Hey @nathzi1505 

Check out the PR I believe it is ready to merge

File: ivy/functional/frontends/tensorflow/general_functions.py [hunk 1]
```diff
@@ -115,6 +115,41 @@ def foldl(
     return result
 
 
+@to_ivy_arrays_and_back
+def foldr(
+    fn,
+    elems,
+    initializer=None,
+    parallel_iterations=10,
+    back_prop=True,
```
Comment (abdulasiraj) [inline ivy/functional/frontends/tensorflow/general_functions.py:124]: why can't we use `back_prop`

**Review by abdulasiraj — CHANGES_REQUESTED** [review]

**Comment by nathzi1505** [comment]
Hi @AbdullahSabry, looks like there are merge conflicts in the commits you made.
Would you mind revising them?

**Comment by AbdullahSabry** [comment]
Hey @nathzi1505 

Just resolved the conflicts in the current commit. Let me know if there's anything to be done!

File: ivy/functional/frontends/tensorflow/general_functions.py (hunk 1 above)
Comment (AbdullahSabry) [reply ivy/functional/frontends/tensorflow/general_functions.py:222]: Not sure how to implement it. Tried using `ivy.execute_with_gradients()` with `retain_grads=True`. However not sure if it serves the same functionality. Let me know what I can do about that!

**Comment by nathzi1505** [comment]
Hi @AbdullahSabry can you please check the lint issues?
Else everything looks good to me!

**Comment by AbdullahSabry** [comment]
Hey @nathzi1505 

Sorry about that, pushed changes to fix it!

**Pull request merged**